### DISCUSSION TITLE
How to suppress warning: undefined global `vim`?

### DISCUSSION THREAD

**Discussion by jojorequiem** [LazyVim/LazyVim#4040] [Q&A]
![undefined_global_vim_lsp](https://github.com/user-attachments/assets/e920a0d6-0052-45ca-95ba-dfdf876dba50)

**Comment by mklauer** [comment]
There was a PR https://github.com/LazyVim/LazyVim/pull/1930 that was rejected as this problem is already handled by lua_ls and neodev in Lazyvim. But for me the warning is still there when I open "lua/config/lazy.lua" from https://github.com/LazyVim/starter, same as in the above screenshot.

**Reply by HeavenTao** [reply]
如果你用的是neovim，那么需要安装[lazydev.nvim](https://github.com/folke/lazydev.nvim)，如果你用的是lazyvim，那么不应该有这个问题

**Reply by mklauer** [reply]
I' have lazyvim and there lazydev.nvim is installed, but I have the problem.

**Comment by gpncarl** [comment]
check if there is a file named `.luarc.json` in the config folder and remove it

**Reply by genamethi** [reply]
(For posterity) Or if you suspect/know it's there intentionally, you can edit the severity level of "undefined-global". This is almost certainly a LuaLS issue. You can take a look at the [LuaLS Settings](https://luals.github.io/wiki/settings/), just make sure you use the json syntax without the "Lua" prefix for settings (see the banner in the link).
